pallets/parachain-system/src/migration.rs

```diff
@@ -34,9 +35,45 @@ pub fn on_runtime_upgrade<T: Config>() -> Weight {
 		StorageVersion::new(1).put::<Pallet<T>>();
 	}
 
+	if StorageVersion::get::<Pallet<T>>() == 1 {
+		weight = weight
+			.saturating_add(v2::migrate::<T>())
+			.saturating_add(T::DbWeight::get().writes(1));
+		STORAGE_VERSION.put::<Pallet<T>>();
+	}
+
 	weight
 }
 
+/// V2: Migrate to 2D weights for ReservedXcmpWeightOverride and ReservedDmpWeightOverride.
+mod v2 {
+	use super::*;
+
+	pub fn migrate<T: Config>() -> Weight {
+		let translate = |pre: u64| -> Weight { Weight::from_parts(pre, MAX_POV_SIZE as u64) };
```

Comment:
do we really want to use `MAX_POV_SIZE` here? is it a limit?

Reply:
It appears so:

https://github.com/paritytech/cumulus/blob/882a892b60e25449903052b946d25bd3a00f30eb/pallets/parachain-system/src/lib.rs#L804-L830

Used as the `max_weight` to `handle_dmp_messages`. It does not seem to be getting any of these numbers from the remaining block weights, so this is set arbitrarily by the developer.

Reply:
But we clearly don't want to use `MAX_POV_SIZE` as the max weight here. When we reach this point of executing the messages, we already don't have the `MAX_POV_SIZE` available anymore.

Reply:
Okay, so let's give it... 64 KB?

Reply:
Sounds reasonable to me.